gcovr/coverage.py

```diff
@@ -398,22 +437,21 @@ def from_covdata(
         subdirs = OrderedDict()
         for key in sorted_keys:
             filecov = covdata[key]
-            DirectoryCoverage.add_directory_coverage(
-                subdirs, filecov.filename, filecov, root_filter
-            )
+            DirectoryCoverage.add_directory_coverage(subdirs, root_filter, filecov)
         return subdirs
 
     @staticmethod
     def directory_key(filename: str, root_filter: re.Pattern):
-        key = os.path.dirname(filename.replace("\\", "/"))
-        if root_filter.search(key + "/") and key != filename:
+        filename = filename.replace("\\", GCOVR_PATH_SEPARATOR)
+        key = os.path.dirname(filename.replace("\\", GCOVR_PATH_SEPARATOR))
+        if root_filter.search(key + GCOVR_PATH_SEPARATOR) and key != filename:
             return key
         return None
 
     @staticmethod
     def directory_root(subdirs: CovData_subdirectories, root_filter: re.Pattern) -> str:
         if not subdirs:
-            return SEPARATOR
+            return GCOVR_PATH_SEPARATOR
```

Comment:
```suggestion
            return "/"
```

Reply:
This should return os.sep, for proper support on Windows.

Reply:
No, it is described in the documents that we use Unix seperators also on Windows.

Reply:
If this was true, a force_unix_separators() call would enclose over this line:
https://github.com/gcovr/gcovr/blob/7111438d5cfdced67dd46c99b2dab7caede0e62f/gcovr/__main__.py#L257
... and on to the corresponding root_filter set on line 275 of __main__.py. Should this also be force_unix_separators() ?

Reply:
Instead, we only force-unix-separators at the time we display on reports and when we sort. Not for managing the data.

Reply:
Since my changes manage filenames to sort through directory management, I need to standardize use of directory-separators how they are managed with the root_filter. If these filenames are coming from os.path.* functions, I also need to use os.sep.

Reply:
See my comment above about normalizing to unix separators in the data model. But after checking the point with the `root_filter` this should be done in a separate PR.